corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js

```diff
@@ -33,6 +33,7 @@ hqDefine("data_dictionary/js/data_dictionary", [
             _.each(groupDict, function (properties, group) {
                 var groupObj = propertyListItem(group, '', true, group, self.name);
                 self.properties.push(groupObj);
+                properties.sort((a, b) => a.index - b.index);
```

Comment:
just confirming how this will work for properties added before the migration.
the value for index would be undefined or null?

Reply:
any new property will have index null?

in terms of ordering, it will be higher to lower i.e highest index value will be on top and 0 or null would be lower in order?

Reply:
The default value for `index` is 0.

Once the user clicks the Save button on data dictionary page, all the properties will be saved with the corresponding indexes from the UI. The ordering is from lower to higher index.

Reply:
> all the properties will be saved with the corresponding indexes from the UI

Right. Where is the change happening in the UI? I was looking for some javascript that would modify the index value for the properties.

> The ordering is from lower to higher index

So lower is on top?

Reply:
> Any old property will have index 0

Since we are not running a migration old values in DB would be nil. Django makes that 0 for the object?

Reply:
> Right. Where is the change happening in the UI? I was looking for some javascript that would modify the index value for the properties. 

The index for the property is getting saved from this piece of [code](https://github.com/dimagi/commcare-hq/pull/32746/files#diff-8b20f9142ae4109168ec759806fe51ff14daaa7eec1f7f18e04c08f0d247e903R136-R147). The `_.each` function is getting used for processing the properties, it provides the data and index for the property and then this `index` is saved to the database. The order in which the properties are in the array corresponds to the ordering that the user sets on UI.

> So lower is on top?

Yes. Lower is on top.

> Since we are not running a migration old values in DB would be nil. Django makes that 0 for the object?

No, we are running a migration to add the `index` field to the database and it has a default value of 0, so 0 will be set as `index` for all the old values that exist before the migration. If we specify a default value for a field, it is also set for all the old values once the migration runs.

Reply:
> it provides the data and index for the property

Cool. So we are using the index given during iteration, and save that as the "index" property.
And when someone reorders, it also reorders in the knockout collection list. Neat!

when someone adds a new case property, is that added on top or in the end?
what is the value of the index for a new case property in the UI?

Reply:
The new property is added at the end. `index` is assigned to the new property during the iteration. We don't need to specify the index and can directly use the index that a property has in the properties array.

Reply:
Super neat!
Thanks